ra2ce/analysis/indirect/losses.py

```diff
@@ -92,228 +117,238 @@ def _check_validity_df(self):
 
-        # shape vlh
-        vlh = pd.DataFrame(
-            index=self.intensities.link_id,  # "link_type"
-        )
-        events = self.criticality_data.filter(regex="^EV")
+        def _create_result(vlh: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+            result = pd.concat(
+                [
+                    vlh,
+                    self.criticality_analysis_non_disrupted[
+                        [f"{self.link_type_column}", "geometry", f"{self.performance_metric}", "detour"] +
+                        list(events.columns)
+                    ]
+                ]
+            )
+            # Get the columns from vehicle_loss_hours that are not in common
+            additional_columns = list(set(vlh.columns) - set(self.criticality_analysis_non_disrupted.columns))
+
+            # Fill 0 for the additional columns of self.criticality_analysis_non_disrupted
+            result.loc[result.index.difference(vlh.index), additional_columns] = result.loc[
+                result.index.difference(vlh.index), additional_columns].fillna(0)
+            return result
+
+        _check_validity_criticality_analysis()
+        _hazard_intensity_ranges = self._get_link_types_heights_ranges()[1]
+        events = self.criticality_analysis.filter(regex=r'^EV(?!1_fr)')
         # Read the performance_change stating the functionality drop
-        performance_change = self.criticality_data[self.performance_metric]
+        performance_change = self.criticality_analysis[self.performance_metric]
+
+        # shape vehicle_loss_hours
+        vehicle_loss_hours_df = pd.DataFrame(columns=['link_id'], data=self.criticality_analysis.index.values)
 
         # find the link_type and the hazard intensity
-        vlh = pd.merge(
-            vlh,
-            self.network[["link_id", "link_type"]],
-            left_index=True,
-            right_on="link_id",
+        vehicle_loss_hours_df = pd.merge(
+            vehicle_loss_hours_df,
+            self.criticality_analysis[
+                [f"{self.link_type_column}", "geometry", f"{self.performance_metric}", "detour"] + list(events.columns)
+                ],
+            left_on="link_id",
+            right_index=True,
         )
-
-        if (
-            self.analysis_type == AnalysisIndirectEnum.MULTI_LINK_LOSSES
-        ):  # only useful for MULTI_LINK_LOSSES
-            vlh = pd.merge(
-                vlh,
-                self.criticality_data[["link_id"] + list(events.columns)],
-                left_index=True,
-                right_on="link_id",
+        vehicle_loss_hours = gpd.GeoDataFrame(
+            vehicle_loss_hours_df, geometry="geometry", crs=self.criticality_analysis.crs)
+        for event in events.columns.tolist():
+            for _, vlh_row in vehicle_loss_hours.iterrows():
+                row_hazard_range = _get_range(eval(vlh_row[event]))
+                row_performance_change = eval(performance_change.loc[vlh_row.link_id])
```

Comment:
Why eval()? My example does not run otherwise. these columns will have float values.